lib/experimental/full-page-client-side-navigation.php

```diff
@@ -0,0 +1,47 @@
+<?php
+/**
+ * Registers full page client-side navigation option using the Interactivity API and adds the necessary directives.
+ */
+
+// Set the navigation mode to full page client-side navigation.
+wp_interactivity_config( 'core/router', array( 'navigationMode' => 'fullPage' ) );
+
+// Enqueue the interactivity router script.
+function _gutenberg_enqueue_interactivity_router() {
+	wp_enqueue_script_module( '@wordpress/interactivity-router' );
+}
+
+add_action( 'init', '_gutenberg_enqueue_interactivity_router', 20 );
+
+// Add directives to all links.
```

Comment:
Same as above, PHPDoc should be used and the params and return values should get documented.

Reply:
Done as part of [this commit](https://github.com/WordPress/gutenberg/pull/59707/commits/36e090173ee052f9d5bbb8a92eca3304e2df2bcb).